File: leetcode2/1easy/최원준/Q496.java
package Leetcode.최원준;

/*
1. 아이디어 :


2. 시간복잡도 :
O( )

3. 자료구조/알고리즘 :

 */

import java.util.HashMap;
import java.util.Map;

public class Q496 {
    class Solution {
        public int[] nextGreaterElement(int[] nums1, int[] nums2) {
            Map<Integer, Integer> biggerNum = new HashMap<>();

            int m = nums2.length;
            for (int i=0; i<m; i++) {
                int num = nums2[i];
                for (int j=i+1; j<m; j++) {
                    int num2 = nums2[j];
                    if (num < num2) {
                        biggerNum.put(num, num2);
                        break;
                    }
                }
            }

            int n = nums1.length;
            int[] ans = new int[n];
            int idx = 0;
            for (int num : nums1) {
                ans[idx++] = biggerNum.containsKey(num) ? biggerNum.get(num) : -1;
            }


            return ans;
        }
    }
}

File: leetcode2/1easy/강병호/Q496.py
class Solution:
    def nextGreaterElement(self, nums1: list[int], nums2: list[int]) -> list[int]:
        result = list()
        for num in nums1:
            check = False
            for idx, num2 in enumerate(nums2):
                print(f"idx :  {idx}")
                if check:
                    if num < num2:
                        result.append(idx)
                    elif idx == len(num2) - 1:
                        result.append(-1)
                
                else:
                    if num == num2:
                        check = True
        
        return result